src/network.rs

```diff
@@ -876,6 +912,29 @@ impl<'a> Network<'a> {
     };
   }
 
+  async fn current_user_saved_albums_contains(&mut self) {
+    let mut app = self.app.lock().await;
+    if let Some(albums) = &app.search_results.albums {
+      let mut album_ids: Vec<String> = Vec::new();
+      albums.albums.items.iter().for_each(|item| {
+        if let Some(id) = &item.id {
+          album_ids.push(id.to_owned());
+        }
+      });
+      if let Ok(are_follwed) = self
+        .spotify
+        .current_user_saved_albums_contains(&album_ids)
+        .await
+      {
+        album_ids.iter().enumerate().for_each(|(i, id)| {
+          if are_follwed[i] {
```

Comment:
Could remove from `saved_album_ids_set` when not followed? Could worth checking if the given album was unfollowed whilst here?